server/spec/activation_key_spec.rb

```diff
@@ -62,12 +62,30 @@
   end
 
   it 'should not allow updating special character as name' do
-    @activation_key['name'] = "fooBar$$"
+    @activation_key['name'] = "foo%%Bar$$"
     owner_client = user_client(@owner, random_string('dummyUser'))
 
-    lambda {
+    expect {
       @activation_key = owner_client.update_activation_key(@activation_key)
-    }.should raise_exception(RestClient::BadRequest)
+    }.to raise_exception(RestClient::BadRequest)
+  end
+
+  it 'should not allow updating empty string as name' do
+    @activation_key['name'] = ""
+    owner_client = user_client(@owner, random_string('dummyUser'))
+
+    expect {
+      @activation_key = owner_client.update_activation_key(@activation_key)
+    }.to raise_exception(RestClient::BadRequest)
+  end
+
+  it 'should not allow updating hypen underscore string as name' do
```

Comment:
Test name describes the opposite of what is being tested. Need to update this to something that indicates it allows hyphens and underscores in names.

Reply:
fixed it.